sqlglot/dialects/fabric.py

```diff
@@ -60,29 +74,35 @@ def datatype_sql(self, expression: exp.DataType) -> str:
             Fabric limits temporal types (TIME, DATETIME2, DATETIMEOFFSET) to max 6 digits precision.
             When no precision is specified, we default to 6 digits.
             """
+            # First apply the parent type mapping to get the correct base type
+            result = super().datatype_sql(expression)
+
+            # Check if this is a temporal type that needs precision handling
             if expression.is_type(
```

Comment:
Let's change this to `if expression.is_type(*exp.DataType.TEMPORAL_TYPES)` instead.